### PR TITLE
refresh slots documentation to diminish emphasis on slot types

### PR DESCRIPTION
**Pull request by joelhawksley**
In their existing state, the slots documentation puts too much emphasis on the "Delegate" and other types that folks generally found confusing.

### CODE REVIEW

File: docs/guide/slots.md [hunk 1]
```diff
@@ -6,28 +6,54 @@ parent: Building ViewComponents
...
+- `renders_many` defines a slot that can be rendered multiple times per-component: `renders_many :items`
 
-_To view documentation for content_areas (deprecated) and the original implementation of Slots (deprecated), see [/content_areas](/content_areas) and [/slots_v1](/slots_v1)._
+For example:
+
+```ruby
+# blog_component.rb
+class BlogComponent < ViewComponent::Base
+  renders_one :header
+  renders_many :posts
```
Comment (Spone) [inline docs/guide/slots.md:22]:
```suggestion
  renders_many :items
```

**Comment by Spone** [comment]
This is a good improvement, every time I had to read this page I got confused about the 3 types of slots :face_with_head_bandage:

File: docs/guide/slots.md [hunk 2]
```diff
@@ -6,28 +6,52 @@ parent: Building ViewComponents
...
+<% posts.each do |post| %>
+  <%= post %>
+<% end %>
+```
 
-Delegate slots delegate to another component:
+```erb
+<%# index.html.erb %>
+<%= render BlogComponent.new do |c| %>
+  <% c.header(classes: '') do %>
```
Comment (Spone) [inline docs/guide/slots.md:42]: @joelhawksley a bit late since you already merged, but I find this `classes: ''` confusing...

Comment (joelhawksley) [reply docs/guide/slots.md:42]: @Spone hmm. What might be a better name?

**Pull request merged**